### PR TITLE
Story 10: Content trigger

### PR DESCRIPTION
**Pull request by Shriyanshagro**
Related to #25 and in continuation to #47

### CODE REVIEW

**Comment by Shriyanshagro** [comment]
Sure, I'll look into those code fragments. I'm not sure but seems like with this code refactoring, I should also bring in implementation for story 11 and 12. This would help me while refactoring.

**Comment by datakurre** [comment]
@Shriyanshagro Regarding some of the currently failing tests, it's good to know that the Plone site id in plone test setup is `plone` in lowercase.

I recall, that name is available for import as a string variable at

```
from plone.app.testing.interfaces import PLONE_SITE_ID
```

that detail is described here https://pypi.org/project/plone.app.testing/

File: src/collective/ifttt/tests/robot/test_content_trigger.robot [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+   Then check 'This will add new IFTTT Trigger' on pagecontent
+
+Scenario: As a site administrator I can configure Content Trigger action
+    Given I'm logged in as a Site Administrator
+    And I fill secret key
+    And I goto home page
+   And I trigger the 'Add IFTTT Content Trigger' action menu item
+   When I fill Content Trigger form
+   Then I see ' Successfully applied the IFTTT event test_event to Plone site' on success
+
```
Comment (datakurre) [inline src/collective/ifttt/tests/robot/test_content_trigger.robot]: @Shriyanshagro Please,

* [ ] fix indentation in these scenarios
* [ ] why there is leading space in ' Successfully ...'

Comment (datakurre) [reply src/collective/ifttt/tests/robot/test_content_trigger.robot]: * [ ]  the current branch contains unnecessary empty file  `src/collective/ifttt/tests/test_content_trigger.py`

File: src/collective/ifttt/utils.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            u'This rule is created to trigger ${ifttt_event_name} '
+            u'on ${title} folder',
+            mapping=dict(
+                ifttt_event_name=data.get('ifttt_event_name'),
+                title=self.context.Title().decode('utf-8', 'ignore'),
+            )
+        )
+
+        # HACK
+        adding = self.context.restrictedTraverse('/Plone/+rule')
```
Comment (datakurre) [inline src/collective/ifttt/utils.py]: @Shriyanshagro You cannot assume that the portal root name is `Plone`.

You can get the portal object 

```python
portal = api.portal.get()
adding = portal.restrictedTraverse('+rule')
```

But possibly that is also available through adapter lookup (faster)

```python
adding = getMultiAdapter((portal, self.request), name='+rule')
```

Comment (datakurre) [reply src/collective/ifttt/utils.py]: Later you use multiadapter lookups, so it is also the right solution here.

This is probably also the reason why acceptance tests fail.

**Comment by datakurre** [comment]
@Shriyanshagro Tests seem to be passing, but there is minor QA issue preventing this to be green

```
/home/travis/build/collective/collective.ifttt/src/collective/ifttt/tests/test_utils.py:8:80: E501 line too long (86 > 79 characters)
```

File: src/collective/ifttt/utils.py [hunk 3]
```diff
@@ -84,7 +87,8 @@ def configure_rule(self, data):
         self.rule_id = storage.values()[-1].id
 
         # traverse to configuration page of content rule
-        rule_url = '/Plone/' + self.rule_id
+        portal = api.portal.get().__name__
```
Comment (datakurre) [inline src/collective/ifttt/utils.py]: Interesting that you found `__name__` for `api.portal.get().__name__`. More usual one found in existing code is `api.portal.get().getId()`.

File: src/collective/ifttt/utils.py [hunk 4]
```diff
@@ -84,7 +87,8 @@ def configure_rule(self, data):
...
+        rule_url = (os.sep).join(('', portal, self.rule_id))
```
Comment (datakurre) [inline src/collective/ifttt/utils.py]: `os.sep` is no no for web paths. Nowadays it is theoretical that someone runs Plone on Windows, but if one does, `os.sep` is `’\`.

Comment (datakurre) [reply src/collective/ifttt/utils.py]: Web is standard, so it should be safe to do `'/'.join()`.

If you want to use a helper, you could use

```
from posixpath import join as urljoin
c = urljoin(a, b)
```
as in https://stackoverflow.com/a/15279799

It uses `/` as separator on every OS.

**Pull request merged**